sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java

```diff
@@ -99,13 +113,148 @@ public SentryEvent process(@NotNull SentryEvent event, @NotNull Hint hint) {
     return transaction;
   }
 
-  private boolean hasAppStartSpan(final @NotNull List<SentrySpan> spans) {
-    for (final SentrySpan span : spans) {
+  private boolean hasAppStartSpan(final @NotNull SentryTransaction txn) {
+    final @NotNull List<SentrySpan> spans = txn.getSpans();
+    for (final @NotNull SentrySpan span : spans) {
       if (span.getOp().contentEquals(APP_START_COLD)
           || span.getOp().contentEquals(APP_START_WARM)) {
         return true;
       }
     }
-    return false;
+
+    final @Nullable SpanContext context = txn.getContexts().getTrace();
+    return context != null
+        && (context.getOperation().equals(APP_START_COLD)
+            || context.getOperation().equals(APP_START_WARM));
+  }
+
+  private void attachColdAppStartSpans(
+      final @NotNull AppStartMetrics appStartMetrics, final @NotNull SentryTransaction txn) {
+
+    // data will be filled anyway only for cold app starts
+    if (appStartMetrics.getAppStartType() != AppStartMetrics.AppStartType.COLD) {
+      return;
+    }
+
+    final @Nullable SpanContext traceContext = txn.getContexts().getTrace();
+    if (traceContext == null) {
+      return;
+    }
+    final @NotNull SentryId traceId = traceContext.getTraceId();
+
+    // Application.onCreate
+    final @NotNull TimeSpan appOnCreate = appStartMetrics.getApplicationOnCreateTimeSpan();
+    if (appOnCreate.hasStopped()) {
+      final SentrySpan span =
+          new SentrySpan(
+              appOnCreate.getStartTimestampS(),
+              appOnCreate.getProjectedStopTimestampS(),
+              traceId,
+              new SpanId(),
+              null,
+              UI_LOAD_OP,
+              appOnCreate.getDescription(),
+              SpanStatus.OK,
+              APP_METRICS_ORIGN,
+              new HashMap<>(),
+              null);
+      txn.getSpans().add(span);
+    }
+
+    // Content Provider
+    final @NotNull List<TimeSpan> contentProviderOnCreates =
+        appStartMetrics.getContentProviderOnCreateTimeSpans();
+    if (!contentProviderOnCreates.isEmpty()) {
+      final @NotNull SentrySpan contentProviderRootSpan =
+          new SentrySpan(
+              contentProviderOnCreates.get(0).getStartTimestampS(),
+              contentProviderOnCreates
+                  .get(contentProviderOnCreates.size() - 1)
+                  .getProjectedStopTimestampS(),
+              traceId,
+              new SpanId(),
+              null,
+              UI_LOAD_OP,
+              "ContentProvider",
+              SpanStatus.OK,
+              APP_METRICS_ORIGN,
+              new HashMap<>(),
+              null);
+      txn.getSpans().add(contentProviderRootSpan);
+      for (final @NotNull TimeSpan contentProvider : contentProviderOnCreates) {
+        final SentrySpan contentProviderSpan =
```

Comment:
this span creation can be moved to a private function, as it's repeated 3 times
same could be done with the root span creation in another separate function